src/Psy/Command/ComposerCommand.php

```diff
@@ -0,0 +1,174 @@
+                new InputArgument('command-name', InputArgument::REQUIRED, ''),
+                new InputArgument('args', InputArgument::IS_ARRAY | InputArgument::OPTIONAL, ''),
+            ))
+            ->setDescription('Composer installation.')
+            ->setHelp(
+                <<<HELP
+composer library Installs the library
+
+if composer is not found will install it locally
+HELP
+            );
+        $this->ignoreValidationErrors();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->input = $input;
+        $this->output = $output;
+
+        if (!$this->checkComposerInstallation()) {
+            $this->installComposer();
+        }
+
+        $this->callComposerBootstrap();
+
+        // extract real command name
+        $tokens = preg_split('{\s+}', $input->__toString());
+        $args = array();
+        foreach ($tokens as $token) {
+            if ($token && $token[0] !== '-') {
+                $args[] = $token;
+                if (count($args) >= 2) {
+                    break;
+                }
+            }
+        }
+        // show help for this command if no command was found
+        if (count($args) < 2) {
+            return parent::run($input, $output);
+        }
+
+        $app = new Application();
+        $app->setAutoExit(false);
+        $input = new StringInput(implode(' ', array_slice($tokens, 1, count($tokens))));
+
+        return $app->doRun($input, $this->output);
+    }
+
+    public function setComposerPath($path)
+    {
+        $this->composerPath = $path;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getSystemShell()
+    {
+        if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
+            return 'cmd.exe';
+        }
+
+        return 'bash';
+    }
+
+    protected function callComposerBootstrap()
+    {
+        // TODO if the path is provided as paramater use it instead
+        require_once 'phar://composer.phar/src/bootstrap.php';
+    }
+
+    /**
+     * @param $command
+     * @return bool|string
+     */
+    protected function shellCommand($command)
+    {
+        $process = proc_open($this->getSystemShell(), $this->specification, $pipes);
+        stream_set_blocking($pipes[self::PIPE_ERR], 0);
+
+        if (is_resource($process)) {
+            fwrite($pipes[self::PIPE_WRITE], $command);
+            fclose($pipes[self::PIPE_WRITE]);
+
+            if ($err = stream_get_contents($pipes[self::PIPE_ERR])) {
+                $err = trim($err);
+                $this->output->writeln("<error>$err</error>");
+
+                return false;
+            }
+
+            $return = stream_get_contents($pipes[self::PIPE_READ]);
+            fclose($pipes[self::PIPE_READ]);
+
+            if (proc_close($process) === 0) {
+                return trim($return);
+            }
+        }
+
+        return false;
+    }
```

Comment:
https://github.com/symfony/Process :)

with this, you do not have to bother anymore with the `proc_open` functions, unix, windows, etc